exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -258,20 +258,27 @@ private void activateService(ServiceInstanceSpec instanceSpec) {
   }
 
   private void stopService(ServiceInstanceSpec instanceSpec) {
-    // TODO: ECR-2334 add restriction for creation new snapshots
     String name = instanceSpec.getName();
     Optional<ServiceWrapper> activeService = findService(name);
     if (activeService.isPresent()) {
       ServiceWrapper service = activeService.get();
-      unRegisterService(service);
+      forbidAccessToBlockchain(service);
       runtimeTransport.disconnectServiceApi(service);
+      unRegisterService(service);
       logger.info("Stopped a service: {}", instanceSpec);
     } else {
       logger.warn("There is no active service with the given name {}. "
           + "Possibly restoring services state after reboot?", name);
     }
   }
 
+  private void forbidAccessToBlockchain(ServiceWrapper service) {
+    Node serviceNode = service.getNode();
```

Comment:
ServiceWrapper exists to provide actions on services as a whole. Requesting a service to stop its operation (currently — to close its Node) is such operation. I'd move this to ServiceWrapper as #close or, better, #requestToStop.